express-notes-api/index.js

```diff
@@ -0,0 +1,91 @@
+// attempting to fix issues where req.body does not equal undefined but it actually equals an empty object when the body is empty????
+const express = require('express');
+const app = express();
+const json = require('./data.json');
+const expressJSON = express.json();
+const fs = require('fs');
+app.use(expressJSON);
+
+app.get('/api/notes', function (req, res) {
+  const array = [];
+  for (const prop in json.notes) {
+    array.push(json.notes[prop]);
+  }
+  res.json(array);
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const id = parseFloat(req.params.id);
+  if (Math.floor(id) !== id || id < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (json.notes[id] === undefined) {
+    res.status(404).json({ error: `cannot find note with id ${id}` });
+  } else if (json.notes[id] !== undefined) {
+    res.status(200).json(json.notes[id]);
+  }
+});
+
+app.post('/api/notes', function (req, res) {
+  if (req.body.content === undefined) {
+    res.status(400).json({ error: 'content is a required field' });
+  } if (req.body !== undefined) {
```

Comment:
If you are using the JSON middleware properly, this condition will never be false.